.github/pull_request_template.md

```diff
@@ -0,0 +1,35 @@
+# Description
+
+Please include a summary of changes, motivation and context for this PR.
+
+Fixes # (issue)
+
+## Type of change
+
+Please tick all options which are relevant.
+
+- [ ] Bug fix
+- [ ] New feature
+- [ ] Requires documentation updates
+
+# Testing
+
+Please describe the test(s) that you added and ran to verify your changes.
+
+## Test(s) description
+
+**Test Configuration**:
+
+* OS:
+
+# Checklist:
+
+- [ ] I have added tests that prove my fix is effective and/or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] I have used understandable variable names
+- [ ] I have squashed my commits into a single commit ready for review
+- [ ] I have run `git-clang-format` on my changes
+- [ ] I have made corresponding changes to the documentation
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] My changes generate no new warnings
```

Comment:
Now our bots will fail if there are new warnings, right? Maybe drop the last bullet?

Reply:
The bots will fail at the moment if there is new warnings, but it was suggested here https://github.com/compiler-research/CppInterOp/pull/290#issuecomment-2124565359 that @maximusron may want to drop that requirement to error on new warnings. I personally think its better to keep the failing on error.

Reply:
I've removed the last bullet point, under the assumption we don't drop -DLLVM_ENABLE_WERROR=On  from the ci.

Reply:
I disagreed there :)